tool/net/lfuncs.c

```diff
@@ -829,6 +829,52 @@ int LuaVisualizeControlCodes(lua_State *L) {
   return LuaCoder(L, VisualizeControlCodes);
 }
 
+int LuaUuidV4(lua_State *L) {
+  char v[] = {'0', '1', '2', '3', '4', '5', '6', '7',
+              '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
+  static char uuid_str[37] = {0};
+
+  uuid_str[0] = v[rand() % 16];
```

Comment:
You probably want to be using `random()` instead of `rand()`, since the latter gives very low-quality randomness (and if the XNU manpage applies to cosmo's implementation, the low dozen bits will just cycle through a pattern rather than being random.)

I think what is actually wanted here is `arc4random` or one of its variants, but those aren't (yet) in cosmopolitan. I'd suggest using `random() % 16` for now.

Reply:
If `random` winds up being expensive, it would also be possible to extract every bit from it to save on the number of calls, but you'd have to have an iteration bound based on `sizeof(long)`…

Reply:
Actually strike that, I guess maybe `random` isn't seeded by default.

You could just use `getrandom`; that's what `GetRandomBytes` is using.